lib/vintage_net/resolver/resolv_conf.ex

```diff
@@ -19,12 +19,18 @@ defmodule VintageNet.Resolver.ResolvConf do
     domains = search_domains(entries)
     name_servers = all_name_servers(entries)
 
-    [Enum.map(domains, &domain_text/1), Enum.map(name_servers, &name_server_text/1)]
+    [
+      "# Generated by VintageNet\n",
+      Enum.map(domains, &domain_text/1),
+      Enum.map(name_servers, &name_server_text/1)
+    ]
   end
 
-  defp domain_text(domain), do: ["search ", domain, "\n"]
+  defp domain_text({ifname, domain}),
+    do: ["# Supplied by #{ifname}\n", "search ", domain, "\n"]
 
-  defp name_server_text(server), do: ["nameserver ", ntoa!(server), "\n"]
+  defp name_server_text({ifname, server}),
+    do: ["# Supplied by #{ifname}\n", "nameserver ", ntoa!(server), "\n"]
```

Comment:
```suggestion
    do: ["nameserver ", VintageNet.IP.ip_to_string(server), " # From ", ifname, "\n"]
```

I didn't look if `ntoa!/1` was used elsewhere, but that's old and I missed replacing it a while back.